src/analysis/retail/warlock/demonology/modules/talents/WickedMaw.tsx

```diff
@@ -0,0 +1,131 @@
+import { formatPercentage, formatThousands } from 'common/format';
+import TALENTS from 'common/TALENTS/warlock';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER, SELECTED_PLAYER_PET } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import Enemies from 'parser/shared/modules/Enemies';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+const WICKED_MAW_DEBUFF_ID = 270569;
+
+class WickedMaw extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+  enemies!: Enemies;
+
+  wickedMawDamage = 0;
+  shadowtouchedDamage = 0;
+  totalDamage = 0;
+
+  wickedMawHits = 0;
+  shadowtouchedHits = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.WICKED_MAW_TALENT);
+
+    if (this.active) {
+      this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onPlayerDamage);
+      // Only track pet damage if shadowtouched is present
+      if (this.selectedCombatant.hasTalent(TALENTS.SHADOWTOUCHED_TALENT)) {
+        this.addEventListener(Events.damage.by(SELECTED_PLAYER_PET), this.onPetDamage);
+      }
+    }
+  }
+
+  onPlayerDamage(event: DamageEvent) {
+    this.totalDamage += event.amount + (event.absorbed || 0);
+
+    const enemy = this.enemies.getEntity(event);
+    if (enemy && enemy.hasBuff(WICKED_MAW_DEBUFF_ID)) {
+      // Calculate Wicked Maw damage increase (20% additional Shadowflame damage to debuffed enemies)
+      const damage = event.amount + (event.absorbed || 0);
+      const baseDamage = damage / 1.2;
+      this.wickedMawDamage += damage - baseDamage;
```

Comment:
can be simplified with the `calculateEffectiveDamage` helper:

```suggestion
      this.wickedMawDamage += calculateEffectiveDamage(event, 0.2);
```

this also needs to be gated by the damage type:

```tsx
const isShadowFlame = isMatchingDamageType(event.ability.type, MAGIC_SCHOOLS.SHADOW.id) || isMatchingDamageType(event.ability.type, MAGIC_SCHOOLS.FIRE.id);
if (!isShadowFlame) {
  return;
}
 // ...
```

(bonus points for making `isMatchingDamageType` work properly for multi-schools passed to the second param, like `isMatchingDamageType(event.ability.type, MAGIC_SCHOOLS.SHADOW.id | MAGIC_SCHOOLS.FIRE.id)`)